torch_xla/experimental/custom_kernel.py

```diff
@@ -1,16 +1,25 @@
 import functools
-import jax
-import jax.numpy as jnp
-import jax._src.pallas.mosaic.pallas_call_registration
 import torch
 import torch_xla
 import torch_xla.core.xla_model as xm
 
-from jax.experimental import pallas as pl
 from typing import List, Callable
 from torch.library import impl
 from torch_xla.core.xla_model import XLA_LIB
 
+
+def jax_import_guard():
+  # Somehow, this could grab the TPU before JAX locks it. Otherwise, any pt-xla TPU operations will hang.
+  xm.xla_device()
```

Comment:
torch_xla._XLAC._init_computation_client()

Reply:
When will this be called in general?

Reply:
we don't usually call this API directly I believe, runtime init usually happens when we try to get the device. However if all you need is to init the runtime this api is cleaner. I used it in https://github.com/pytorch/pytorch/blob/a04e7fca8eddde492b239da6ac23d6a056666a0e/benchmarks/dynamo/common.py#L91 as well

Reply:
would this API work correctly on a multipod environment? @JackCaoG

Reply:
hmm it should. We just need to make sure that PyTorch/XLA init the runtime first and grab the libtpu, JAX init the runtime after pytorch doesn't cause any issue on single pod for us at least. We don't use JAX to execute any device program so it is OK. 

In mutipod it sounds like init the runtime twice will cause issue? I never look into that too much

Reply:
@jonb377 any thoughts on why this issue may happen on multi-pod?